security/advancedtls/crl_provider_test.go

```diff
@@ -191,9 +197,11 @@ func (s) TestFileWatcherCRLProvider(t *testing.T) {
 			}
 		})
 	}
+	nonCRLMutex.Lock()
```

Comment:
Does calling `p.Close` here, before the check, not remove the need for the mutex?  If not, why not?  I would think that would eliminate the concurrency concerns.

Reply:
Yeah that's a very good point - current `p.Close` impl waits for gouroutines to end. I tested locally and `-race` is green